service/src/main/java/org/cbioportal/genome_nexus/service/internal/CancerHotspotServiceImpl.java

```diff
@@ -156,25 +159,29 @@ public List<Hotspot> getHotspotAnnotationsByGenomicLocation(String genomicLocati
         throws VariantAnnotationNotFoundException, VariantAnnotationWebServiceException,
         CancerHotspotsWebServiceException
     {
-        GenomicLocation location = this.notationConverter.parseGenomicLocation(genomicLocation);
-
-        return this.getHotspotAnnotationsByVariant(this.notationConverter.genomicToHgvs(location));
+        VariantAnnotation variantAnnotation = genomicLocationAnnotationService.getAnnotation(genomicLocation);
+        return this.getHotspotAnnotations(variantAnnotation);
     }
 
     @Override
     public List<AggregatedHotspots> getHotspotAnnotationsByGenomicLocations(List<GenomicLocation> genomicLocations)
         throws CancerHotspotsWebServiceException
     {
-        // convert genomic location to hgvs notation (there is always 1-1 mapping)
-        Map<String, GenomicLocation> variantToGenomicLocation = notationConverter.genomicToHgvsMap(genomicLocations);
+        List<VariantAnnotation> variantAnnotations = this.genomicLocationAnnotationService.getAnnotations(genomicLocations);
 
-        // query hotspots service by variant
-        List<AggregatedHotspots> hotspots = this.getHotspotAnnotationsByVariants(
-            new ArrayList<>(variantToGenomicLocation.keySet()));
-
-        // add genomic location info too
-        for (AggregatedHotspots aggregatedHotspots: hotspots) {
-            aggregatedHotspots.setGenomicLocation(variantToGenomicLocation.get(aggregatedHotspots.getVariant()));
+        Function<GenomicLocation, String> genomicLocationToVariantFormat = this.genomicLocationAnnotationService.getGenomicLocationToVariantFormat();
+        Map<String, GenomicLocation> variantToGenomicLocation = new HashMap<>();
+        for (GenomicLocation genomicLocation: genomicLocations) {
+            variantToGenomicLocation.put(genomicLocationToVariantFormat.apply(genomicLocation), genomicLocation);
+        }
+        List<AggregatedHotspots> hotspots = new ArrayList<>();
+        for (VariantAnnotation variantAnnotation : variantAnnotations)
+        {
+            AggregatedHotspots aggregatedHotspots = new AggregatedHotspots();
+            aggregatedHotspots.setHotspots(this.getHotspotAnnotations(variantAnnotation));
+            aggregatedHotspots.setVariant(variantAnnotation.getVariant());
```

Comment:
if `getVariant()` is different in this case, we might be breaking something. Let's check this.

Reply:
I've checked that the variant field does not seem to be used anywhere really. Maybe it's worth removing altogether?
@sheridancbio Is genomenexus api used by cbioportal only? What'd be a procedure to introduce a non-backward compatible change?

Reply:
The swagger says that the api is "internal" on this page https://www.genomenexus.org/swagger-ui.html

I am not sure what exactly it means in this context.

Reply:
@ruslan-forostianov The two main uses of the genomenexus api (in particular the annotation endpoint) are:
1. the cBioPortal website uses it for dynamic data requests during the rendering of certain pages (such as the mutation viewer / lollypop display)
2. the MSK importer codebase, and related batch annotation tools used by the data curators call the genomenexus api to annotate genomic events. We try to "pre-annotate" our files so that loading occurs faster. We also have a nightly fetch pipeline which annotates  the events for the samples added to the MSK cBioPortal site, and that fetch pipeline uses genome-nexus. But if an un-annotated mutations file gets imported, the the annotation will be done at time of import. These calls all go through the genome nexus client library in https://github.com/genome-nexus/genome-nexus-annotation-pipeline

There may be other institutions which use the genome-nexus api, but I am not aware of a specific case. We did introduce changes which were (I think) not backwards compatible a few years back. I think I remember that we added a parameter to annotation requests which was a maximum execution time (after which the job would auto-cancel). This might have been in the genome-nexus-vep codebase however. We added code to always report the original query string when responding to a request for annotation, I think. (although maybe adding a field to the response still could be considered backwards compatible). But anyway, I believe we are free to make adjustments to the API when it makes sense. cBioPortal (and related importers) must be the dominant user of the system. My opinion is that API-breaking changes could be made, so long as a transition plan for rolling them out is agreed on by all the main clients. However, there are many installations of cBioPortal outside of the core development teams, so if the change would disrupt function of the cBioPortal frontend codebase we would probably need to version the API so that the older API would still be available. Other developers of genome-nexus should comment on this too.